src/sync/sync_user.cpp

```diff
@@ -55,14 +55,15 @@ static std::vector<std::string> split_token(const std::string& jwt) {
     return parts;
 }
 
-RealmJWT::RealmJWT(const std::string& token)
+RealmJWT::RealmJWT(std::string& token)
```

Comment:
This is tricky because callers may pass in their token variable without explicitly moving it and have the contents moved unknowingly. Accepting a variable by reference does technically have the right to modify the variable in any way, but it is usually better to be explicit about having a `f(T&& v)` so that callers can opt in to this efficient no-copy function if they don't intend to use the variable later. It is not uncommon to implement two versions of the constructor though, one that moves and one that copies so users have the flexibility to choose by being explicit about `std::move`. So I'd recommend we have two constructors `RealmJWT(std::string&& token)` and `RealmJWT(const std::string& token)` which will make a copy when assigned to `this->token`. I'm being pedantic here because it can lead to some hard to track down bugs if memory is moved around without explicitly using `std::move` at the call site, because the code reads like it is correct until you look at the function definition and then you'd have to track down all calls to the function and verify they are using it correctly.

Reply:
I've chosen to only have `RealmJWT(std::string&& token)` I don't think there is any need to have 2 constructors at the moment considering there are very few codepaths initialising a `RealmJWT`